src/Components/CommunityListShelf.js

```diff
@@ -0,0 +1,103 @@
+import Typography from "@mui/material/Typography";
+import Box from "@mui/material/Box";
+import { getAvatarSrc } from "./Avatars";
+import { Fragment, useEffect, useMemo, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useProfile } from "./APICalls";
+import Avatar from "@mui/material/Avatar";
+
+import AnimeShelf from "./AnimeShelf";
+import Skeleton from "@mui/material/Skeleton";
+import { useMediaQuery } from "@mui/material";
+
+export default function CommunityListShelf({ data, anime, titleStyle }) {
+  const navigate = useNavigate();
+  const { data: profile, isLoading: profileLoading } = useProfile(data?.userId);
+  const cannotHover = useMediaQuery("(hover: none)");
+  const coarsePointer = useMediaQuery("(pointer: coarse)");
+
+  const isTouchscreen = cannotHover && coarsePointer;
+
+  const avatarSrc = useMemo(
+    () => getAvatarSrc(profile?.avatar),
+    [profile?.avatar]
+  );
+
+  const listName = useMemo(() => {
```

Comment:
Nice use of `useMemo`!

Reply:
Thanks! Writing this function really helped me understand it better. Here `useMemo` is definitely much easier to read/understand than having to chase around a separate state!